pept/scanners/parallel_screens/extensions/read_adac_binary.pyx

```diff
@@ -0,0 +1,246 @@
+	cdef double itag = 0
+	cdef double itagold = 0
+	cdef int nevent = 0
+
+	cdef int Gx1=590
+	cdef int Gx2=590
+	cdef int Gy1=590
+	cdef int Gy2=590
+
+	cdef unsigned int word
+	cdef unsigned char word3
+	cdef unsigned char word2
+	cdef unsigned char word1
+	cdef unsigned char word0
+
+	cdef double ix
+	cdef double ix1
+	cdef double ix2
+	cdef double iy
+	cdef double iy1
+	cdef double iy2
+
+	LORs = []
+
+	if saveas == None:
+		saveas = fname[:-5]+".csv"
+
+	# cdef char* fname_C = fname
+	# cdef char* saveas_C = saveas
+
+	# read_adac_binary_ext(fname_C, saveas_C)
+
+	f_out = open(saveas, "w")
+
+	with open(fname,"rb") as f:
+
+		if separation == None:
+			separation = int.from_bytes(f.read(2),'little')
+		else:
+			f.seek(2)
+		gantry_angle = int.from_bytes(f.read(2),'little')
+		header = f.seek(1000) # skip the header
+
+		# Read data words until LOR is complete
+
+
+		word_bytes = f.read(4)
+
+		# while(word!=:
+
+		while(  word_bytes != b'' ):
+			# while(GotA and GotB and GotXA and GotXB and GotYA and GotYB) == 0:
+
+			word_bytes = f.read(4)
+			word = int.from_bytes(word_bytes,'little')
+
+			if hex(word) == '0xfaceface':
+				# print(hex(word))
+				# print("Skipped that one\n")
+				word_bytes = f.read(4)
+				word = int.from_bytes(word_bytes,'little')
+
+
+			word = word&0xFFFFFFA0
+
+			itag = ((word&0xFF000000) >> 24)
+			dtime = itag - itagold
+			itagold = itag
+
+			if(dtime<0):
+				dtime = dtime + 256
+			if(dtime<130):
+				itime = itime + (dtime / TimeStep)
+			
+			prev_word = word
+
+			word3 = (word&0xFF000000) >> 24 
+			word2 = (word&0x00FF0000) >> 16 
+			word1 = (word&0x0000FF00) >> 8 
+			word0 = (word&0x000000FF) 
+
+			# parse data value from inverted and odd order word
+			value = 16383 - ((word2&16)//16) -((word2&64)//32) - ((word2&2)*2) - ((word1&128)//16) - ((word1&32)//2) - ((word1&8)*4) - ((word2&4)*16) - ((word2&8)*16) - ((word2&32)*8) - ((word2&1)*512) - ((word1&64)*16) - ((word1&16)*128) - ((word1&4)*1024) - ((word1&1)*8192)
+	
+			# event data
+
+			ix=value
+			iy=value
+
+			# get word ident values
+			
+			TrgX = 1 - ((word&0x80)>>7)
+			TrgY = 1 - ((word&0x200)>>9)
+			TrgZ = (0b1 - TrgX)&(0b1-TrgY)
+
+			TrgH = 1 - ((word&0x800000)>>23)
+			TrgP = (word&0x400000)>>22
+
+			# if(TrgX):
+			# 	print("X");
+			# if(TrgY):
+			# 	print("Y");
+			# if(TrgZ):
+			# 	print("Z");
+			# if(TrgH):
+			# 	print("H");
+			# if(TrgP):
+			# 	print("P");
+			# print("dtime = ", dtime, " itag= ", itime, " TimeStep = ", TimeStep)
+			# print("itime = ", itime)
+
+			# input('')
+			
+			# X word
+			if(TrgX):
+				# X pair 
+				if(TrgP):
+					pair = 1
+					GotB = 1
+				else:
+					pair = 0
+					GotA=1
+					GotB=0
+					GotXA=0
+					GotXB=0
+					GotYA=0
+					GotYB=0
+
+			if(TrgX):
+				if(TrgH==0):
+					ix1 = ix/16
+					GotXA=1
+				else:
+					ix2 = ix/16
+					GotXB=1
+
+
+
+			if(TrgY):
+				if(TrgH==0):
+					iy1 = 1024-(1024-(iy/16))
+					GotYA=1
+				else:
+					iy2 = 1024 - (iy/16)
```

Comment:
I know I am being pedantic, but:
- Could we try to add spaces between operators (=, -, /, ==)? 
- Code can be shorter if using this initialisation form: x = y = z = d = 0

Reply:
I tried that initialization form with cdef and Cython didn't like it